drivers/gpio/gpio_rpi_pico.c

```diff
@@ -29,32 +29,52 @@ struct gpio_rpi_data {
 	struct gpio_driver_data common;
 	sys_slist_t callbacks;
 	uint32_t int_enabled_mask;
+	uint32_t single_ended_mask;
+	uint32_t open_drain_mask;
 };
 
 static int gpio_rpi_configure(const struct device *dev,
 				gpio_pin_t pin,
 				gpio_flags_t flags)
 {
-	if (flags & GPIO_SINGLE_ENDED) {
-		return -ENOTSUP;
-	}
+	struct gpio_rpi_data *data = dev->data;
+
+	gpio_set_pulls(pin,
+		(flags & GPIO_PULL_UP) != 0U,
+		(flags & GPIO_PULL_DOWN) != 0U);
 
 	/* Avoid gpio_init, since that also clears previously set direction/high/low */
 	gpio_set_function(pin, GPIO_FUNC_SIO);
 
 	if (flags & GPIO_OUTPUT) {
-		gpio_set_dir(pin, GPIO_OUT);
-
-		if (flags & GPIO_OUTPUT_INIT_HIGH) {
-			gpio_put(pin, 1);
-		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
-			gpio_put(pin, 0);
+		if (flags & GPIO_SINGLE_ENDED) {
+			data->single_ended_mask |= 1 << pin;
+			if (flags & GPIO_LINE_OPEN_DRAIN) {
+				data->open_drain_mask |= 1 << pin;
+				gpio_put(pin, 0);
+			} else {
+				data->open_drain_mask &= ~(1 << pin);
+				gpio_put(pin, 1);
+			}
+			if (((flags & GPIO_LINE_OPEN_DRAIN) &&
+				(flags & GPIO_OUTPUT_INIT_LOW)) ||
+				((!(flags & GPIO_LINE_OPEN_DRAIN)) &&
+				(flags & GPIO_OUTPUT_INIT_HIGH))) {
+				gpio_set_dir(pin, GPIO_OUT);
+			} else {
+				gpio_set_dir(pin, GPIO_IN);
```

Comment:
```
gpio_set_dir(pin, ((flags & GPIO_LINE_OPEN_DRAIN) &&
			  (flags & GPIO_OUTPUT_INIT_LOW)) ||
			 ((!(flags & GPIO_LINE_OPEN_DRAIN)) &&
			  (flags & GPIO_OUTPUT_INIT_HIGH)));
```
instead of that if/else is a bit cleaner, and also gets rid of an if (gpio_set_dir has an if of its own internally)

Reply:
DONE, Not sure I fully agree for readability, cause now the code requires people to look up the signature of gpio_set_dir to know what the value means, but I made the change anyway.

Reply:
How about:
```
if (flags & GPIO_LINE_OPEN_DRAIN) {
	data->open_drain_mask |= BIT(pin);
	gpio_put(pin, 0);
	gpio_set_dir(pin, (flags & GPIO_OUTPUT_INIT_LOW));
} else {
	data->open_drain_mask &= ~(BIT(pin));
	gpio_put(pin, 1);
	gpio_set_dir(pin, (flags & GPIO_OUTPUT_INIT_HIGH));
}
```
?
Normally a `!!` would have been needed, but not in this instance because `gpio_set_dir` has an internal if.

Reply:
DONE.  Does it really matter if there's a check or not in gpio_set_dir? I though using bool as an argument type would collapse the value to 0/1. And even if bool was not a real C99 stdbool, but just an int masquerading as a bool on some non-standard compiler, I would still consider it a bug if a function taking something marked as bool would treat it as a bitmask, so I wouldn't have expected gpio_set_dir to have problems.